encoding/da.go

```diff
@@ -269,3 +269,54 @@ func TxsToTxsData(txs types.Transactions) []*types.TransactionData {
 	}
 	return txsData
 }
+
+// Fast testing if the compressed data is compatible with our circuit
+// (require specified frame header and each block is compressed)
+func CheckCompressedDataCompatibility(data []byte) error {
```

Comment:
Do we also need to add this check after trying to compress every batch for early return?

Reply:
As mentioned above:

>can apply this method when the uncompressed batch data has exceed 128kB (important: never apply it when data is less than 128kB).

When the size of all uncompressed batches exceed 128kB, apply this check would bring an additional ensurance

Reply:
so could i add this check in this pr?

Reply:
added in [01c9be9](https://github.com/scroll-tech/da-codec/pull/16/commits/01c9be9125aad91d0052fafdd4b76ad5cb8490cf), pls review, `codecv3` also uses this function.